libs/pilight/protocols/433.92/rev_v5.c

```diff
@@ -0,0 +1,414 @@
+	createLow(4,47);
+}
+
+static int createIdFromLetter(int l) {
+	int i=0;
+
+	for(i=0;i<(ID_COUNT-1);i++) {
+		if((int)idLetters[i] == l) { //lookup the position of the char
+            return i;
+		}
+	}
+
+	//Char not found
+    return -1;
+}
+
+static void createId(int id) {
+	//copied from rev_v3.c
+	int binary[255];
+	int length = 0;
+	int i=0, x=0;
+
+	length = decToBinRev(id, binary);
+	for(i=0;i<=length;i++) {
+		x=i*4;
+		if(binary[i]==1) {
+			createHigh(x, x+3);
+		}
+	}
+}
+
+static int generateUnitValue(int unit, int areabit) {
+	const int UNITMSBMASK = 0x02;
+	const int UNITLSBMASK = 0x01;
+	int result = 0;
+	int unitMsb = (unit & UNITMSBMASK) << 2;
+
+	areabit = areabit << 2;
+              /*1st*/          /*2nd*/             /*3rd*/        /*4th*/
+	result =  unitMsb | /*2nd bit is always 0 |*/ areabit | (unit & UNITLSBMASK);
+
+	return result;
+}
+
+static void createUnit(int unit) {
+	//based on rev_v3.c
+	int binary[255];
+	int length = 0;
+	int i=0, x=0;
+
+	//logprintf(LOG_INFO, "rev_v5:sending unit value '%d'", unit);
+
+	length = decToBinRev(unit, binary);
+	for(i=0;i<=length;i++) {
+		if(binary[i]==1) {
+			x=i*4;
+			createHigh(16+x, 16+x+3); //first unit pulse is 16
+		}
+	}
+}
+
+
+static void createState(bool on, bool all) {
+	//based on rev_v3.c
+	int binary[255];
+	int length = 0;
+	int i=0, x=0;
+    int rawState;
+
+	if(all) {
+		rawState = on ? 0 : 3; //ALL-ON -> 0 //ALL-OFF -> 3
+	}
+	else {
+		rawState = on ? 1 : 2; //ON -> 1 //OFF -> 2
+	}
+	
+	length = decToBinRev(rawState, binary);
+
+	//logprintf(LOG_INFO, "rev_v5:sending state value '%d', bit-length:%d", rawState, length);
+
+	for(i=0;i<=length;i++) {
+		if(binary[i]==1) {
+			x=i*4;
+			createHigh(32+x, 32+x+3); //first unit pulse is 32
+		}
+	}
+}
+
+static void createPostfix(void) {
+	//the last 2 bits are always set
+	createHigh(40, 40+3); 
+	createHigh(44, 44+3); 
+}
+
+static void createFooter(void) {
+	//copied from rev_v3.c
+	rev5_switch->raw[48]=(AVG_PULSE_LENGTH);
+	rev5_switch->raw[49]=(PULSE_DIV*AVG_PULSE_LENGTH);
+}
+
+static int createCode(struct JsonNode *code) {
+	char sId[4] = {'\0'};
+	char *stmp = NULL;
+	double itmp = -1;
+	int id = -1;
+	int unit = -1;
+	int state = -1;
+	int all = 0;
+	int areabit = 0;
+
+	strcpy(sId, "-1");
+
+	if(json_find_string(code, "id", &stmp) == 0)
+		strcpy(sId, stmp);
+
+	if((int)(sId[0]) >= 48 /*0*/ && (int)(sId[0]) <= 57 /*9*/) {
+		//Id is already a number
+		id = atoi(sId);
+	}
+    else if((int)(sId[0]) >= 65 /*A*/ && (int)(sId[0]) <= 80 /*P*/) {
+		//Id is defined by char
+		id = createIdFromLetter((int)sId[0]);
+	}
+	else {
+		logprintf(LOG_ERR, "rev5_switch: invalid id number specified");
+	}
+	
+
+	if(json_find_number(code, "unit", &itmp) == 0)
+		unit = (int)round(itmp);
+	if(json_find_number(code, "all", &itmp) == 0)
+		all = (int)round(itmp);
+	if(json_find_number(code, "off", &itmp) == 0)
+		state=0;
+	if(json_find_number(code, "on", &itmp) == 0)
+		state=1;
```

Comment:
Inconsistent style.

Reply:
I think you mean the missing curly braces{}, otherwise you had written Indentation.

Reply:
Yes